content/en/apps/examples/covid-rdt-reference-app.md

```diff
@@ -0,0 +1,490 @@
+ * **3rd party integrations:** We have used both [Klipfolio](https://www.klipfolio.com/) and, more recently, [Superset](https://superset.incubator.apache.org/) to create more user-friendly dashboards than any of the above options offer.  They talk to the Postgres server as the back end.
+
+
+### Sample API call
+
+All calls to the API need to be authenticated. Be sure to follow our documentation on how to do this, but otherwise you need to use a login with `admin` permissions with this stucture: `http://LOGIN:PASSWORD@HOSTNAME`. If you had deployed this application without any customizations, you would have these forms available as seen by calling the [forms API]({{< ref "apps/reference/api#get-apiv1forms" >}}) (`/api/v1/forms`):
+
+* contact:clinic:create.xml
+* contact:clinic:edit.xml
+* contact:district_hospital:create.xml
+* contact:district_hospital:edit.xml
+* contact:health_center:create.xml
+* contact:health_center:edit.xml
+* contact:person:create.xml
+* contact:person:edit.xml
+* covid19_rdt_capture.xml
+* covid19_rdt_provision.xml
+
+The last two forms can be used to query [the reports API ]({{< ref "apps/reference/api#get-apiv2exportreports" >}}) to get RDT provions and captures.  To get all the capture reports into a file called `output.csv`, you would use this call:
+
+```shell
+curl "http://LOGIN:PASSWORD@HOSTNAME/api/v2/export/reports?filters[search]=&filters[forms][selected][0][code]=covid19_rdt_capture" > output.csv
+```
+
+### Base64 image extraction
+
+To extract the binary image from the ASCII value in a report, you can view it in the CHT or you use a script to do so.  Here's an example of using Bash utilities (`sed`, `cut`, `tr` and `base64`) in Linux to get the image in the 1st row of data after the column headers into the file `image.jpg`. 
+
+```shell
+sed '2q;d' output.csv | cut -d',' -f15 | tr -d '"' | base64 -d > image.jpg
+```
+
+When retrieving [JSON](#capture-1), this value is found in `capture.android-app-outputs.rdt_session_bundle.rdt_session_result_bundle.rdt_session_result_extra_images.cropped` field.
+
+{{% alert title="Note" %}}The reports API always outputs in CSV, but the COVID-19 application uses [Base64 encoding](https://en.wikipedia.org/wiki/Base64) to store the images as text as noted above.  These may misbehave when opened them in a spreadsheet application like LibreOffice ("maximum number of characters per cell exceeded") or Excel (silently clipped to 32k chars) as they're thousands, if not hundreds of thousands, of characters long.  Be sure to programmatically process these into image files as needed. {{% /alert %}}
+
+
+## Directory structure
+
+Shown below is a filtered file structure of the COVID-19 app where you'll want to focus your customization efforts:
```

Comment:
It wasn't clear to me what the "Directory structure" header was referring to, or what the filtered file structure criteria was. Here is a suggested adjustment:

```suggestion
## Customizing the reference app

These are the files in the COVID-19 app where you'll want to focus your customization efforts:
```